src/engine/strat_engine/thinpool.rs

```diff
@@ -55,16 +66,62 @@ pub struct ThinPoolStatus {
 
 impl ThinPool {
     /// Make a new thin pool.
-    #[allow(too_many_arguments)]
     pub fn new(pool_uuid: PoolUuid,
                dm: &DM,
                data_block_size: Sectors,
                low_water_mark: DataBlocks,
-               spare_segments: Vec<Segment>,
-               meta_dev: LinearDev,
-               data_dev: LinearDev,
-               mdv: MetadataVol)
+               block_mgr: &mut BlockDevMgr)
                -> EngineResult<ThinPool> {
+        if block_mgr.avail_space() < ThinPool::initial_size() {
+            let avail_size = block_mgr.avail_space().bytes();
+            return Err(EngineError::Engine(ErrorEnum::Invalid,
+                                           format!("Space on pool must be at least {} bytes, \
```

Comment:
This is minor, but I'ld prefer it if you assigned the error message to an identifier, say ```err_msg```, and used that as an argument. All this crowding right is a bit annoying.